src/cryptonote_core/miner.cpp

```diff
@@ -365,6 +398,11 @@ namespace cryptonote
         misc_utils::sleep_no_w(100);
         continue;
       }
+      else if( m_is_background_mining_enabled )
+      {
+        misc_utils::sleep_no_w(m_miner_extra_sleep);
```

Comment:
I missed that there was a separate configuration option for targetting CPU level. The description explains it clearly, but the internal name is similar to the idle threshold value. Instead of variations on "mining threshold", how about variations of "mining target"? I hate to do this nitpick, but I think it would help with code clarity a bit (and its a search + replace to do).

As for the mutex idea, don't do it if you are not comfortable implemented the logic for it. Updating to the lock method should not be a difficult refactor.